### PR TITLE
[NET-5314] Limit v2 Service port registration to L4 TCP ports

### PR DESCRIPTION
**Pull request by zalimeni**
Ignore non-TCP L4 ports in K8s services. This is expected behavior and also prevents unintended duplication of Service port values registered to Consul (which is not supported) when ports have multiplexed L4 traffic.

Changes proposed in this PR:
- Filter Service port registrations to L4 TCP protocol ports
- Retain debug logging that helped identify this issue

How I've tested this PR: Added a unit test and made it pass

How I expect reviewers to test this PR: 👀 

Checklist:
- [x] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: control-plane/connect-inject/controllers/endpointsv2/endpoints_controller_test.go [hunk 1]
```diff
@@ -169,17 +169,17 @@ func TestReconcile_CreateService(t *testing.T) {
 							Addresses: addressesForPods(pod1, pod2),
 							Ports: []corev1.EndpointPort{
 								{
-									Name:        "my-http-port",
+									Name:        "public",
```
Comment (zalimeni) [inline control-plane/connect-inject/controllers/endpointsv2/endpoints_controller_test.go:172]: `ServicePort` and `EndpointPort` names are supposed to match in K8s; noticed these were incorrect so fixing along w/ added test.

File: control-plane/connect-inject/controllers/endpointsv2/endpoints_controller.go [hunk 2]
```diff
@@ -202,23 +203,24 @@ func getOwnerPrefixFromPod(pod *corev1.Pod) string {
...
 	)
 
-	r.Log.Info("registering service with Consul", getLogFieldsForResource(serviceResource.Id)...)
+	r.Log.Info("registering service with Consul", getLogFieldsForResource(consulSvcResource.Id)...)
 	//TODO: Maybe attempt to debounce redundant writes. For now, we blindly rewrite state on each reconcile.
-	_, err := resourceClient.Write(ctx, &pbresource.WriteRequest{Resource: serviceResource})
+	_, err := resourceClient.Write(ctx, &pbresource.WriteRequest{Resource: consulSvcResource})
 	if err != nil {
-		r.Log.Error(err, "failed to register service", getLogFieldsForResource(serviceResource.Id)...)
+		r.Log.Error(err, fmt.Sprintf("failed to register service: %+v", consulSvc), getLogFieldsForResource(consulSvcResource.Id)...)
```
Comment (zalimeni) [inline control-plane/connect-inject/controllers/endpointsv2/endpoints_controller.go:223]: This logging seemed valuable to keep around. There's no secrets in this value, so I think it's reasonable to include. Happy to remove if folks disagree

File: control-plane/connect-inject/controllers/endpointsv2/endpoints_controller.go [hunk 3]
```diff
@@ -254,13 +256,21 @@ func getServicePorts(service corev1.Service) []*pbcatalog.ServicePort {
...
-			TargetPort: p.TargetPort.String(),
-			Protocol:   common.GetPortProtocol(p.AppProtocol),
-		})
+		// Service mesh only supports TCP as the L4 Protocol (not to be confused w/ L7 AppProtocol).
+		//
+		// This check is necessary to deduplicate VirtualPort values when multiple declared ServicePort values exist
+		// for the same port, which is possible in K8s when e.g. multiplexing TCP and UDP traffic over a single port.
+		//
+		// If we otherwise see repeat port values in a K8s service, we pass along and allow Consul to fail validation.
+		if p.Protocol == "TCP" {
```
Comment (ishustava) [inline control-plane/connect-inject/controllers/endpointsv2/endpoints_controller.go:265]: nit: wondering if there's some constant in k8s libraries for this?

**Review by ishustava — APPROVED** [review]
LGTM

File: control-plane/connect-inject/controllers/endpointsv2/endpoints_controller.go (hunk 3 above)
Comment (zalimeni) [reply control-plane/connect-inject/controllers/endpointsv2/endpoints_controller.go]: Ah, somehow missed [this](https://github.com/kubernetes/kubernetes/blob/7c42715e57979a1c3cc065f3e3914c0ef3caf779/pkg/apis/core/types.go#L762) the first time I checked! Updated, good callout.

**Comment by zalimeni** [comment]
Rebased on `main` test fix, no changes

**Pull request merged**